server/src/main/java/org/elasticsearch/health/RestGetHealthAction.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.health;
+
+import org.elasticsearch.client.internal.node.NodeClient;
+import org.elasticsearch.rest.BaseRestHandler;
+import org.elasticsearch.rest.RestRequest;
+import org.elasticsearch.rest.action.RestToXContentListener;
+
+import java.io.IOException;
+import java.util.List;
+
+import static org.elasticsearch.rest.RestRequest.Method.GET;
+
+public class RestGetHealthAction extends BaseRestHandler {
+
+    @Override
+    public String getName() {
+        // TODO: Existing - "cluster_health_action", "cat_health_action"
+        return "health_action";
+    }
+
+    @Override
+    public List<Route> routes() {
+        return List.of(new Route(GET, "/_internal/_health"));
```

Comment:
Have we settled on this as the path for this API? Sorry if I missed the discussion on it. I don't understand why we're including the term `internal`.

Reply:
We want to expose it there early to enable early consumers to use it. The `_internal` namespace signals internal to Elastic use only as well as no-bwc expectation for outside consumers. It also allows us to completely depart the health endpoint, in case we find something completely different is necessary.

Reply:
:+1: ok. If we intend to remove this API should we mark it as deprecated already? I don't think we have a way to return warning headers indicating that an API is experimental, although that might be useful, but deprecating it straight away would have a similar effect.

Reply:
Let me know what you all want here. I am fine with whatever approach.

Reply:
I'd prefer not to deprecate it initially. I think that could lead to unforeseen issues like deprecation warnings logged due to calling this new API.